public/pages/Dashboard/Container/DashboardOverview.tsx

```diff
@@ -157,14 +191,24 @@ export function DashboardOverview() {
   };
 
   const intializeDetectors = async () => {
-    dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS));
-    dispatch(getIndices(''));
-    dispatch(getAliases(''));
+    dispatch(getDetectorList(getAllDetectorsQueryParamsWithDataSourceId(MDSOverviewState.selectedDataSourceId)));
+    dispatch(getIndices('', MDSOverviewState.selectedDataSourceId));
+    dispatch(getAliases('', MDSOverviewState.selectedDataSourceId));
   };
 
   useEffect(() => {
-    intializeDetectors();
-  }, []);
+    const { history, location } = props;
+    const updatedParams = {
+      dataSourceId: MDSOverviewState.selectedDataSourceId,
+    };
+    history.replace({
+      ...location,
+      search: queryString.stringify(updatedParams),
+    })
+    if (dataSourceEnabled ? MDSOverviewState.selectedDataSourceId : true) {
+      intializeDetectors();
+    }
```

Comment:
The example you provided will result in initializeDetectors() not being called if MDS is not enabled. When MDS is enabled, selectedDataSourceId should always exist because the data source picker provides a default data source when rendering. If not, the data source picker will display an error banner. Additionally, later in the reducer, I initialized dataSourceId as '' if it doesn't have a truthy value, allowing it to fall back to the local cluster case. Therefore, I think we can remove the if-check here and trust the process we implement later.